database/schema.sql

```diff
@@ -0,0 +1,55 @@
+-- ----------------------------------------------------------------
+-- Users
+-- ----------------------------------------------------------------
+CREATE TYPE USER_ROLE AS ENUM (
+    'user',
+    'it'
+);
+
+CREATE TABLE IF NOT EXISTS users (
+    id              SERIAL PRIMARY KEY                       UNIQUE,
+    firstname       VARCHAR(64)                              NOT NULL,
+    lastname        VARCHAR(64)                              NOT NULL,
+    email           VARCHAR(64)                              NOT NULL,
+    managerID       REFERENCES users (id) ON DELETE CASCADE,
+    role            USER_ROLE                                NOT NULL DEFAULT 'user',
+    username        VARCHAR(64)                              UNIQUE NOT NULL,
+    password        VARCHAR(64)                              NOT NULL
+);
+
+-- ----------------------------------------------------------------
+-- Tickets
+-- from professor feedback, no archive table. simply have query
+-- to return closed tickets
+-- ----------------------------------------------------------------
+CREATE TYPE TICKET_STATUS AS ENUM (
+    'open',
+    'in_progress',
+    'closed'
+);
+
+CREATE TYPE ISSUE_TYPE AS ENUM (
+    'other',
+    'another issue type',
+    'another issue type'
+);
```

Comment:
Are these just placeholder names? Maybe we could have just simply, hardware/software/other